In [39]:
import thinkstats2
import nsfg
import math

In [5]:
%who

thinkstats2	 


In [6]:
thinkstats2

<module 'thinkstats2' from '/Users/alexandrezajic/metis/prework/ThinkStats2/code/thinkstats2.py'>

In [11]:
df = nsfg.ReadFemPreg()

In [19]:
df.columns

Index(['caseid', 'pregordr', 'howpreg_n', 'howpreg_p', 'moscurrp', 'nowprgdk',
       'pregend1', 'pregend2', 'nbrnaliv', 'multbrth',
       ...
       'laborfor_i', 'religion_i', 'metro_i', 'basewgt', 'adj_mod_basewgt',
       'finalwgt', 'secu_p', 'sest', 'cmintvw', 'totalwgt_lb'],
      dtype='object', length=244)

In [20]:
df['totalwgt_lb'] = df['birthwgt_lb'] + (df['birthwgt_oz'] / 16)

In [28]:
df['totalwgt_lb']

0        8.8125
1        7.8750
2        9.1250
3        7.0000
4        6.1875
5        8.5625
6        9.5625
7        8.3750
8        7.5625
9        6.6250
10       7.8125
11       7.0000
12       4.0000
13          NaN
14          NaN
15       7.6875
16       7.5000
17       6.3125
18          NaN
19       8.7500
20       8.1875
21       5.5625
22          NaN
23       6.7500
24       7.3750
25       6.8125
26       8.1250
27       7.1250
28       6.0625
29       7.4375
          ...  
13563    7.6875
13564    7.6250
13565    8.1250
13566    7.5000
13567       NaN
13568       NaN
13569    5.8125
13570    6.6875
13571    6.0000
13572    5.8125
13573    6.5625
13574    6.1250
13575       NaN
13576    6.4375
13577       NaN
13578    6.0000
13579    7.0000
13580       NaN
13581    6.3750
13582       NaN
13583       NaN
13584    6.3750
13585       NaN
13586       NaN
13587       NaN
13588    6.1875
13589       NaN
13590       NaN
13591    7.5000
13592    7.5000
Name: totalwgt_lb, Lengt

In [ ]:
df['totalwgt_lb'].isna() == df['birthwgt_lb'].isna()
# Eyeballing quickly that no error in my calculation was causing new "NaN"s to be created

In [35]:
df['pregordr']

0        1
1        2
2        1
3        2
4        3
5        1
6        2
7        3
8        1
9        2
10       1
11       1
12       2
13       3
14       1
15       2
16       3
17       1
18       2
19       1
20       2
21       1
22       2
23       1
24       2
25       3
26       1
27       1
28       2
29       3
        ..
13563    2
13564    3
13565    1
13566    1
13567    1
13568    2
13569    1
13570    2
13571    3
13572    4
13573    1
13574    2
13575    1
13576    1
13577    2
13578    1
13579    2
13580    1
13581    2
13582    3
13583    1
13584    2
13585    1
13586    2
13587    3
13588    1
13589    2
13590    3
13591    4
13592    5
Name: pregordr, Length: 13593, dtype: int64

In [36]:
first_babies = df.loc[df['pregordr'] == 1, 'totalwgt_lb']
later_babies = df.loc[df['pregordr'] != 1, 'totalwgt_lb']

In [45]:
def CohensD(pop1, pop2):
    mean_diff = abs(pop1.mean() - pop2.mean())
    
    n1, n2 = len(pop1), len(pop2)
    var1 = pop1.var()
    var2 = pop2.var()
    
    pooled_var = (n1 * var1 + n2 * var2)/(n1 + n2)
    d = mean_diff / math.sqrt(pooled_var)
    return d

In [46]:
CohensD(first_babies, later_babies)

0.06911825348820934

Cohen's D for **birth weight** differences between first babies and others is ~0.069, which is noticeably higher than Cohen's D for **pregnancy length** differences between first babies and others (~0.029).

Despite this difference between the Cohen's D values, Cohen's D for birth weight differences is still quite small, at 0.069 standard deviations of variation.

*Though this calculation comes from a sample, I chose to stay consistent with his earlier calculations and calculate the variance as he did, without factoring in the differences between samples and populations*